dev/com.ibm.ws.microprofile.health.2.0_fat/fat/src/com/ibm/ws/microprofile/health20/fat/ApplicationStateHealthCheckTest.java

```diff
@@ -70,235 +85,216 @@
-        WebArchive dynamicApp1 = ShrinkHelper.buildDefaultApp(DIFFERENT_APP_NAME, "com.ibm.ws.microprofile.health20.different.app.name.health.checks.app");
-        ShrinkHelper.exportDropinAppToServer(server1, dynamicApp1);
-
-        log("testDynamicallyLoadedDropinsHealthCheckTest", "Waiting for DifferentAppNameHealthCheckApp to start");
-        server1.waitForStringInLog("CWWKZ0001I.* " + DIFFERENT_APP_NAME, APP_STARTUP_TIMEOUT);
+    public void testDynamicallyLoadedApplicationsHealthCheckTest() throws Exception {
+        server1.setServerConfigurationFile("delayed-server.xml");
+        log("testDynamicallyLoadedApplicationsHealthCheckTest", "Starting the server and dynamically adding " + SUCCESSFUL_APP_NAME);
+        loadServerAndApplication(SUCCESSFUL_APP_NAME, "com.ibm.ws.microprofile.health20.successful.health.checks.app", true);
 
-        log("testDynamicallyLoadedDropinsHealthCheckTest", "Testing health check endpoints after DifferentAppNameHealthCheckApp is dynamically deployed");
+        log("testDynamicallyLoadedApplicationsHealthCheckTest", "Testing health check endpoints after " + SUCCESSFUL_APP_NAME + " is dynamically deployed");
         expectHealthCheck(HealthCheck.LIVE, Status.SUCCESS, 1);
-        expectHealthCheck(HealthCheck.READY, Status.FAILURE, 1);
-        expectHealthCheck(HealthCheck.HEALTH, Status.FAILURE, 3);
+        expectHealthCheck(HealthCheck.READY, Status.SUCCESS, 1);
+        expectHealthCheck(HealthCheck.HEALTH, Status.SUCCESS, 2);
 
-        log("testDynamicallyLoadedDropinsHealthCheckTest", "Adding MultipleHealthCheckApp to dropins");
-        WebArchive dynamicApp2 = ShrinkHelper.buildDefaultApp(MULTIPLE_APP_NAME, "com.ibm.ws.microprofile.health20.multiple.health.checks.app");
-        ShrinkHelper.exportDropinAppToServer(server1, dynamicApp2);
+        log("testDynamicallyLoadedApplicationsHealthCheckTest", "Adding " + DELAYED_APP_NAME + " to dropins");
+        exportApplication(DELAYED_APP_NAME, "com.ibm.ws.microprofile.health20.delayed.health.check.app");
 
-        log("testDynamicallyLoadedDropinsHealthCheckTest", "Waiting for MultipleHealthCheckApp to start");
-        server1.waitForStringInLog("CWWKZ0001I.* " + MULTIPLE_APP_NAME, APP_STARTUP_TIMEOUT);
+        log("testDynamicallyLoadedApplicationsHealthCheckTest", "Testing for readiness DOWN while " + DELAYED_APP_NAME + " is starting.");
+        try {
+            HttpURLConnection conReady = null;
+            int responseCode = -1;
+            boolean first_time = true;
+            boolean app_ready = false;
+            long start_time = System.currentTimeMillis();
+            long time_out = 180000; // 180000ms = 3min
+
+            // Repeatedly hit the readiness endpoint until an UP response is received
+            while (!app_ready) {
+                conReady = HttpUtils.getHttpConnectionWithAnyResponseCode(server1, READY_ENDPOINT);
```

Comment:
Repeatedly hitting the endpoint that is not ready yet, might throw some ConnectException/SocketTimeoutException/SocketException, so we have to catch that and continue to keep trying it. Refer to the test case (https://github.com/OpenLiberty/open-liberty/blob/integration/dev/com.ibm.ws.microprofile.health.2.0_fat/fat/src/com/ibm/ws/microprofile/health20/fat/DelayAppStartupHealthCheckTest.java#L150)

Reply:
Resolved. Shouldn't need this since there is an expectHealthCheck call beforehand which would only pass if the server had successfully started.